torch/distributed/_composable/fsdp/fully_shard.py

```diff
@@ -17,13 +19,27 @@
 def fully_shard(
     module: nn.Module,
     *,
+    mesh: Optional[DeviceMesh] = None,
     device: DeviceLikeType = "cuda",
 ):
     if isinstance(module, (nn.ModuleList, nn.ModuleDict)):
         raise ValueError(
             f"fully_shard does not support containers that do not implement forward: {module}"
         )
     device = _normalize_device(device)
+    mesh = mesh or _init_default_fully_shard_mesh(device.type)
+    if mesh.ndim not in (1, 2):
+        raise ValueError(f"fully_shard expects a 1D or 2D DeviceMesh but got {mesh}")
+    elif mesh.ndim == 1:
+        mesh_info = FSDPMeshInfo(mesh, shard_mesh_dim=0)
```

Comment:
are we going to support ``DDPMeshInfo`` where there is only ``replicate_mesh_dim`` but no ``shard_mesh_dim`` ?

Reply:
I do not think we are planning to do that in the near term. It is just used for `HSDPMeshInfo` for now.